### PR TITLE
reference implementation of BIP44 for

### PR DESCRIPTION
**Pull request by buckysballs**

### CODE REVIEW

**Comment by buckysballs** [comment]
Hey @coranos I just made a reference implementation. We'll add some more utils/to the cli but you can use this for testing

**Comment by coranos** [comment]
thanks. I'll try to review it this weekend.

File: wallet/src/main/scala/org/constellation/wallet/BIP44.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.bouncycastle.asn1.sec.SECNamedCurves
+import org.bouncycastle.asn1.x9.X9ECParameters
+import org.bouncycastle.jce.spec.{ECParameterSpec, ECPrivateKeySpec, _}
+import org.bouncycastle.math.ec
+import org.constellation.keytool.KeyUtils
+import org.constellation.keytool.KeyUtils.insertProvider
+import org.constellation.wallet.BIP44.{getPrivateKeyFromECBigIntAndCurve, getPublicKeyFromECPoint}
+
+class BIP44(seedPrase: String, childIndex: Int = 0, passphrase: String = "", creationTime: Long = new Date().getTime) {
+  val chainPathPrefix: String = "M/44H/137036H/0H/0/" //137.036 is the fine structure constant ...
```
Comment (TheMMaciek) [inline wallet/src/main/scala/org/constellation/wallet/BIP44.scala:19]: Isn't the coin_type level supposed to be and ID for DAG from SLIP-44? So:
1137 | 0x80000471

File: wallet/src/main/scala/org/constellation/wallet/BIP44.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+}
+
+object BIP44 extends App {
+  import org.bouncycastle.jce.ECNamedCurveTable
+
+  def getPrivateKeyFromECBigIntAndCurve(s: BigInteger, curveName: String) = {
+    val ecParameterSpec: ECParameterSpec = ECNamedCurveTable.getParameterSpec(curveName)
+    val privateKeySpec: ECPrivateKeySpec = new ECPrivateKeySpec(s, ecParameterSpec)
+    val keyFactory = KeyFactory.getInstance("ECDSA")
```
Comment (TheMMaciek) [inline wallet/src/main/scala/org/constellation/wallet/BIP44.scala:47]: According to: https://docs.oracle.com/javase/7/docs/technotes/guides/security/StandardNames.html#KeyFactory
there is no "ECDSA" algorithm supplied, I guess "EC" is what we need.

Comment (TheMMaciek) [reply wallet/src/main/scala/org/constellation/wallet/BIP44.scala:47]: @marcinwadon explained that it needs inserting the BouncyCastleProvider just like in getPublicKeyFromECPoint

File: wallet/src/main/scala/org/constellation/wallet/BIP44.scala (hunk 1 above)
Comment (buckysballs) [reply wallet/src/main/scala/org/constellation/wallet/BIP44.scala:19]: Must've forgotten that I already registered our coin type, yep it should be 1137

Comment (TheMMaciek) [reply wallet/src/main/scala/org/constellation/wallet/BIP44.scala]: I pushed one commit changing coin type value to 1137

**Pull request closed without merging**